grunt/config/replace.js

```diff
@@ -37,6 +37,18 @@ module.exports = function (grunt, options) {
         if (configJson._xapi.hasOwnProperty('_activityID') && configJson._xapi._activityID == '') {
           grunt.log.writeln('WARNING: xAPI activityID has not been set');
         }
+
+        // cmi5 is a profile of the xAPI specification and some systems, e.g. SCORM Cloud
+        // do not work well with both manifest types, so remove the one which is not being used.
+        if (configJson._xapi.hasOwnProperty('_specification')) {
```

Comment:
i wanted to keep it as far away from grunt as possible, seeing as i'd like to get rid of grunt at some point. 
i quite like the npm scripts object in the package.json, which is why my code is analogous. https://docs.npmjs.com/files/package.json#scripts also .bowerrc for the bower-cli has something similar https://bower.io/docs/config/#hooks but the bower.json doesn't https://github.com/bower/spec/blob/master/json.md so we're quite free to add our own.

i would go with:

1. a ``scripts/`` folder
2. a ``scripts: {}`` object in the bower.json
```json
{
  "name": "adapt-contrib-spoor",
  "version": "2.1.1",
  "framework": "^2.0.16",
  "homepage": "https://github.com/adaptlearning/adapt-contrib-spoor",
  "issues": "https://github.com/adaptlearning/adapt_framework/issues/new?title=contrib-spoor%3A%20please%20enter%20a%20brief%20summary%20of%20the%20issue%20here&body=please%20provide%20a%20full%20description%20of%20the%20problem,%20including%20steps%20on%20how%20to%20replicate,%20what%20browser(s)/device(s)%20the%20problem%20occurs%20on%20and,%20where%20helpful,%20the%20contents%20of%20the%20SCORM%20debug%20window.",
  "displayName" : "Spoor",
  "extension" : "spoor",
  "description": "An extension which allows basic SCORM tracking",
  "main": "/js/adapt-contrib-spoor.js",
  "scripts": {
      "adaptprebuild": "scripts/copy.js",
      "adaptbuild": "scripts/replace.js",
      "adaptpostbuild": "scripts/cleanup.js"
   },
  "keywords": [
    "adapt-plugin",
    "adapt-extension"
  ],
  "license": "GPLv3",
  "targetAttribute": "_spoor"
}
```  
2. have a defined api for those scripts, to be a module defined like:
```
module.exports = function(fs, path, log, options) {
    // where
    options = {
        sourcedir: "/path/to/root/src",
        outputdir: "/path/to/root/build",
        plugindir: "/path/to/root/src/extension/adapt-contrib-spoor"
    };
}
```

that way the plugins can run any node based code, for any number of src+build combinations, without needing to know of grunt.
like:

```javascript
module.exports = function(fs, path, log, options) {

  var readJSON = function(path) {
    return JSON.parse(fs.readFileSync(path).toString());
  };
  var writeJSON = function(path, json) {
    return fs.writeFileSync(path, JSON.stringify(json, null, 4));
  };

  var configJson = readJSON(pathToConfig);
  var defaultLanguage = configJson._defaultLanguage || 'en';
  var courseJson = readJSON(path.join(courseDir, defaultLanguage, 'course.json'));
  
  var hasChanged = false;

  // Backwards compatibility for courses missing 'description'
  if (!courseJson.hasOwnProperty('description')) {
    courseJson.description = '';
    hasChanged = true;
  }

  // A shim for edge cases where xAPI has not been configured.
  if (!configJson.hasOwnProperty('_xapi')) {
    configJson._xapi = {
      '_activityID': '',
      '_isEnabled': false
    };
    hasChanged = true;
  } 

  if (hasChanged) {
    writeJSON(pathToConfig, configJson);
  }

  if (configJson._isEnabled) {
    // xAPI has been enabled, check if the activityID has been set.
    if (configJson._xapi.hasOwnProperty('_activityID') && configJson._xapi._activityID == '') {
      log('WARNING: xAPI activityID has not been set');
    }

    // cmi5 is a profile of the xAPI specification and some systems, e.g. SCORM Cloud
    // do not work well with both manifest types, so remove the one which is not being used.
    if (configJson._xapi.hasOwnProperty('_specification')) {
      if (configJson._xapi._specification === 'xAPI') {
        // Remove the cmi5.xml file.
        var cmi5Path = path.join(options.outputdir, 'cmi5.xml')
        if (fs.fileExistsSync(cmi5Path)) fs.unlink(cmi5Path);
      } else if (configJson._xapi._specification === 'cmi5') {
        // Remove the tincan.xml file.
        var tincanPath = path.join(options.outputdir, 'tincan.xml');
        if (fs.fileExistsSync(tincanPath)) fs.unlink(tincanPath);
      }
    }
  }

};
```